### PR TITLE
Fix links to the Eta Expansion section in the spec

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

**Review by diesalbla — APPROVED** [review]

**Comment by diesalbla** [comment]
@lrytz Any reason why this documentation PR should wait? Looks harmless.

**Comment by dwijnand** [comment]
`#good-to-go-after-2.13.0-RC2` 😉

**Review by hrhino — CHANGES_REQUESTED** [review]
I'm not so sure this is the right solution: the problem is that this section steals the "#eta-expansion" anchor, whereas what you really want is the explanation of how eta-expansion works (which lives at the bottom of the page). 

I tried to fix this before and failed miserably, because the Jekyll docs don't explain if there's a way to customise or suppress anchors. Nor could I figure out a decent way to produce a `<dl>`, which would also probably work. (A normal list with the bolded title and a colon up front, maybe, but that offends my unsensible sensibilities).

This isn't the only link to `#eta-expansion`, by the way, and they all go here rather than there. (Now it looks like I'm coining a hashtag.)

File: spec/06-expressions.md [hunk 1]
```diff
@@ -1746,7 +1746,7 @@ a = scala.Any
 
 so `scala.Any` is the type inferred for `a`.
 
-### Eta Expansion
+### <a name="eta-expansion-section">Eta Expansion</a>
```
Comment (hrhino) [inline spec/06-expressions.md:1749]: !

well that was obvious

**Review by hrhino — APPROVED** [review]

File: spec/06-expressions.md (hunk 1 above)
Comment (dwijnand) [reply spec/06-expressions.md:1749]: Apparently, it's implementation specific, so we'll see if it works.

https://stackoverflow.com/a/6494918/463761

Comment (hrhino) [reply spec/06-expressions.md:1749]: As the kids say, 🤞.

**Comment by som-snytt** [comment]
Thanks, this is very helpful. Two things I dread are when I have to consult the javadoc for java.lang.String yet again, and also chase down this section of the scala spec yet again.

There is a possible missed opportunity to have called it the `eta-expansion-expansion`, since that is what is expanded upon.

**Pull request merged**